packages/speech/src/index.js

```diff
@@ -0,0 +1,816 @@
+ *
+ * @example
+ * var config = {
+ *   encoding: 'LINEAR16',
+ *   sampleRate: 16000
+ * };
+ *
+ * function callback(err, transcript, apiResponse) {
+ *   if (err) {
+ *     // Error handling omitted.
+ *   }
+ *
+ *   // transcript = "how old is the Brooklyn Bridge"
+ * }
+ *
+ * //-
+ * // <h3>Run speech detection over a local file</h3>
+ * //-
+ * speech.recognize('./bridge.raw', config, callback);
+ *
+ * //-
+ * // <h3>Run speech recognition over a file in Cloud Storage</h3>
+ * //-
+ * speech.recognize('gs://your-bucket-name/bridge.raw', config, callback);
+ *
+ * //-
+ * // <h3>Run speech recognition over raw file contents</h3>
+ * //-
+ * speech.recognize({
+ *   content: fs.readFileSync('./bridge.raw')
+ * }, config, callback);
+ *
+ * //-
+ * // <h3>Run speech recognition over a remote file</h3>
+ * //
+ * // *Note: This is not an officially supported feature of the Speech API.
+ * // google-cloud will make a request to the URL given and send the file
+ * // contents to the upstream API.*
+ * //-
+ * speech.recognize('https://example.com/files/bridge.raw', config, callback);
+ *
+ * //-
+ * // <h3>Enable verbose mode for more detailed results.</h3>
+ * //-
+ * var config = {
+ *   encoding: 'LINEAR16',
+ *   sampleRate: 16000,
+ *   verbose: true
+ * };
+ *
+ * speech.recognize('./bridge.raw', config, function(err, results) {
+ *   if (err) {
+ *     // Error handling omitted.
+ *   }
+ *
+ *   // results = [
+ *   //   {
+ *   //     transcript: "how old is the Brooklyn Bridge",
+ *   //     confidence: 88.15,
+ *   //     alternatives: [
+ *   //       {
+ *   //         transcript: "how old is the Brooklyn brim",
+ *   //         confidence: 22.39
+ *   //       }
+ *   //     ]
+ *   //   }
+ *   // ]
+ * });
+ */
+Speech.prototype.recognize = function(file, config, callback) {
+  var self = this;
+
+  var protoOpts = {
+    service: 'Speech',
+    method: 'syncRecognize'
+  };
+
+  config = extend({}, config);
+
+  if (!config.encoding) {
+    config.encoding = Speech.detectEncoding_(file);
+  }
+
+  var verboseMode = config.verbose === true;
+  delete config.verbose;
+
+  Speech.findFile_(file, function(err, foundFile) {
+    if (err) {
+      callback(err);
+      return;
+    }
+
+    var reqOpts = {
+      audio: foundFile,
+      config: config
+    };
+
+    self.request(protoOpts, reqOpts, function(err, apiResponse) {
+      if (err) {
+        callback(err, null, apiResponse);
+        return;
+      }
+
+      var response = new self.protos.Speech.SyncRecognizeResponse(apiResponse);
+      var results = Speech.formatResults_(response.results, verboseMode);
+
+      callback(null, results, apiResponse);
+    });
+  });
+};
+
+/**
+ * Perform asynchronous speech recognition.
+ *
+ * This method sends audio to the Speech API, which immediately responds with an
+ * Operation object. You can then poll the operation for a result by registering
```

Comment:
I think we can safely remove the part about polling, this is something that happens behind the scenes so the user shouldn't need to concern themselves with it. All we should tell them is that they'll need to listen from the `complete` event.